### PR TITLE
Add option to run the project with visible collision shapes and navigation

### PR DESCRIPTION
**Pull request by was-games**
See issue #287 

- **run_project_debug** command: runs the project with visible collision shapes and navigation meshed
- **force_visible_collision_shapes** option: forces visible collision shapes when running the project normally
- **fforce_visible_nav_mesh** option: forces visible navigation meshes when running the project normally

resolved:
Promises' type error

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -308,4 +322,4 @@
 		"vscode-languageclient": "^5.2.1",
 		"ws": "^7.4.6"
 	}
-}
+}
```
Comment (Calinou) [inline package.json:325]:
```suggestion
}

```

**Comment by Calinou** [comment]
@was-games This needs to be rebased before it can be merged, could you look into it? Thanks in advance :slightly_smiling_face:

**Comment by was-games** [comment]
@Calinou Is everything OK now?

**Comment by Calinou** [comment]
> @Calinou Is everything OK now?

There are still merge conflicts to resolve according to GitHub.

**Comment by was-games** [comment]
@Calinou Thank you for your patience :). 

I think there is a problem with a current master branch. Below is the output when trying to run it. Let me check if I can make it run before rebasing.

```
	"resource": "/C:/Users/avakhtel.FINE/Documents/Projects/godot-vscode-plugin/src/godot-tools.ts",
	"owner": "typescript",
	"code": "2339",
	"severity": 8,
	"message": "Property 'stop' does not exist on type 'GDScriptLanguageClient'.",
	"source": "ts",
	"startLineNumber": 48,
	"startColumn": 15,
	"endLineNumber": 48,
	"endColumn": 19
```

```
	"resource": "/C:/Users/avakhtel.FINE/Documents/Projects/godot-vscode-plugin/src/lsp/GDScriptLanguageClient.ts",
	"owner": "typescript",
	"code": "2307",
	"severity": 8,
	"message": "Cannot find module 'vscode-languageclient/node' or its corresponding type declarations.",
	"source": "ts",
	"startLineNumber": 2,
	"startColumn": 86,
	"endLineNumber": 2,
	"endColumn": 114
```

```
[{
	"resource": "/C:/Users/avakhtel.FINE/Documents/Projects/godot-vscode-plugin/src/lsp/MessageIO.ts",
	"owner": "typescript",
	"code": "2724",
	"severity": 8,
	"message": "'\"vscode-jsonrpc\"' has no exported member named 'AbstractMessageReader'. Did you mean 'StreamMessageReader'?",
	"source": "ts",
	"startLineNumber": 1,
	"startColumn": 10,
	"endLineNumber": 1,
	"endColumn": 31
},{
	"resource": "/C:/Users/avakhtel.FINE/Documents/Projects/godot-vscode-plugin/src/lsp/MessageIO.ts",
	"owner": "typescript",
	"code": "2724",
	"severity": 8,
	"message": "'\"vscode-jsonrpc\"' has no exported member named 'AbstractMessageWriter'. Did you mean 'StreamMessageWriter'?",
	"source": "ts",
	"startLineNumber": 7,
	"startColumn": 10,
	"endLineNumber": 7,
	"endColumn": 31
},{
	"resource": "/C:/Users/avakhtel.FINE/Documents/Projects/godot-vscode-plugin/src/lsp/MessageIO.ts",
	"owner": "typescript",
	"code": "2459",
	"severity": 8,
	"message": "Module '\"vscode-jsonrpc\"' declares 'ResponseMessage' locally, but it is not exported.",
	"source": "ts",
	"startLineNumber": 8,
	"startColumn": 26,
	"endLineNumber": 8,
	"endColumn": 41,
	"relatedInformation": [
		{
			"startLineNumber": 3,
			"startColumn": 201,
			"endLineNumber": 3,
			"endColumn": 216,
			"message": "'ResponseMessage' is declared here.",
			"resource": "/C:/Users/avakhtel.FINE/Documents/Projects/godot-vscode-plugin/node_modules/vscode-jsonrpc/lib/main.d.ts"
		}
	]
},{
	"resource": "/C:/Users/avakhtel.FINE/Documents/Projects/godot-vscode-plugin/src/lsp/MessageIO.ts",
	"owner": "typescript",
	"code": "2420",
	"severity": 8,
	"message": "Class 'MessageIOReader' incorrectly implements interface 'MessageReader'.\n  Type 'MessageIOReader' is missing the following properties from type 'MessageReader': onError, onClose, onPartialMessage, dispose",
	"source": "ts",
	"startLineNumber": 104,
	"startColumn": 14,
	"endLineNumber": 104,
	"endColumn": 29
},{
	"resource": "/C:/Users/avakhtel.FINE/Documents/Projects/godot-vscode-plugin/src/lsp/MessageIO.ts",
	"owner": "typescript",
	"code": "2339",
	"severity": 8,
	"message": "Property 'fireError' does not exist on type 'MessageIOReader'.",
	"source": "ts",
	"startLineNumber": 138,
	"startColumn": 44,
	"endLineNumber": 138,
	"endColumn": 53
},{
	"resource": "/C:/Users/avakhtel.FINE/Documents/Projects/godot-vscode-plugin/src/lsp/MessageIO.ts",
	"owner": "typescript",
	"code": "2339",
	"severity": 8,
	"message": "Property 'fireClose' does not exist on type 'MessageIOReader'.",
	"source": "ts",
	"startLineNumber": 139,
	"startColumn": 34,
	"endLineNumber": 139,
	"endColumn": 43
},{
	"resource": "/C:/Users/avakhtel.FINE/Documents/Projects/godot-vscode-plugin/src/lsp/MessageIO.ts",
	"owner": "typescript",
	"code": "2339",
	"severity": 8,
	"message": "Property 'firePartialMessage' does not exist on type 'MessageIOReader'.",
	"source": "ts",
	"startLineNumber": 195,
	"startColumn": 10,
	"endLineNumber": 195,
	"endColumn": 28
},{
	"resource": "/C:/Users/avakhtel.FINE/Documents/Projects/godot-vscode-plugin/src/lsp/MessageIO.ts",
	"owner": "typescript",
	"code": "2420",
	"severity": 8,
	"message": "Class 'MessageIOWriter' incorrectly implements interface 'MessageWriter'.\n  Type 'MessageIOWriter' is missing the following properties from type 'MessageWriter': onError, onClose, dispose",
	"source": "ts",
	"startLineNumber": 204,
	"startColumn": 14,
	"endLineNumber": 204,
	"endColumn": 29
},{
	"resource": "/C:/Users/avakhtel.FINE/Documents/Projects/godot-vscode-plugin/src/lsp/MessageIO.ts",
	"owner": "typescript",
	"code": "2339",
	"severity": 8,
	"message": "Property 'fireError' does not exist on type 'MessageIOWriter'.",
	"source": "ts",
	"startLineNumber": 216,
	"startColumn": 44,
	"endLineNumber": 216,
	"endColumn": 53
},{
	"resource": "/C:/Users/avakhtel.FINE/Documents/Projects/godot-vscode-plugin/src/lsp/MessageIO.ts",
	"owner": "typescript",
	"code": "2339",
	"severity": 8,
	"message": "Property 'fireClose' does not exist on type 'MessageIOWriter'.",
	"source": "ts",
	"startLineNumber": 217,
	"startColumn": 34,
	"endLineNumber": 217,
	"endColumn": 43
},{
	"resource": "/C:/Users/avakhtel.FINE/Documents/Projects/godot-vscode-plugin/src/lsp/MessageIO.ts",
	"owner": "typescript",
	"code": "2339",
	"severity": 8,
	"message": "Property 'fireError' does not exist on type 'MessageIOWriter'.",
	"source": "ts",
	"startLineNumber": 242,
	"startColumn": 9,
	"endLineNumber": 242,
	"endColumn": 18
}]
```

**Comment by was-games** [comment]
This was just npm being npm and not working as it should.

**Comment by was-games** [comment]
@Calinou sorry for the spamy behavior. No conflicts now.

**Review by Calinou — APPROVED** [review]
Thanks!

**Pull request merged**